picard/ui/options/plugins.py

```diff
@@ -257,10 +257,13 @@ def add_plugin_item(self, plugin, item=None):
             button.setMaximumHeight(button.fontMetrics().boundingRect(label).height() + 7)
             self.ui.plugins.setItemWidget(item, 2, button)
 
-            def download_button_process():
+            def download_button_process(action):
                 self.ui.plugins.setCurrentItem(item)
-                self.download_plugin()
-            button.released.connect(download_button_process)
+                self.download_plugin(action)
+            if label == _("Update"):
```

Comment:
Don't do that. Instead mark string for translation, using `N_()`, call `_()` the later possible, and define constants for possible actions. 

```python

PLUGIN_ACTION_NONE, PLUGIN_ACTION_UPDATE, PLUGIN_ACTION_INSTALL = range(3)

...

bt_action = PLUGIN_ACTION_NONE
 if plugin.can_be_updated:
    bt_action = PLUGIN_ACTION_UPDATE
 elif plugin.can_be_downloaded:
    bt_action = PLUGIN_ACTION_INSTALL
    item.setFlags(item.flags() ^ QtCore.Qt.ItemIsUserCheckable)

if bt_action != PLUGIN_ACTION_NONE:
    labels = {
      PLUGIN_ACTION_UPDATE:  N_("Update"),
      PLUGIN_ACTION_INSTALL: N_("Install"),
    }
    button = QtWidgets.QPushButton(_(labels[bt_action]))
  
...
    button.released.connect(partial(download_button_process, bt_action))
...


def download_handler(self, action, response, reply, error, plugin):

    ...

    if action == PLUGIN_ACTION_INSTALL:

        ...

```

Instead of calling functions and comparing strings, you compare integers, and don't call useless functions.